### PR TITLE
Fix: make sql queries compatible with sqlite

### PR DESCRIPTION
**Pull request by sharafian**
As per https://github.com/interledgerjs/ilp-kit/issues/150 .

Some of the functions or SQL commands that were used were only compatible with postgresql. This turns those into forms that are allowed by both sqlite and postgres.

This allows an ILP Kit to be set up very quickly, without going through the lengthy process of getting postgresql working on your machine.

### CODE REVIEW

File: api/src/models/payment.js [hunk 1]
```diff
@@ -93,8 +93,8 @@ function PaymentFactory (sequelize, validator, container, User) {
           + 'destination_identifier, destination_name, destination_image_url,'
           + ' message, time_slot'
         + ' ORDER BY recent_date DESC'
-        + ' OFFSET ' + limit * (page - 1)
-        + ' LIMIT ' + limit,
+        + ' LIMIT ' + limit
+        + ' OFFSET ' + limit * (page - 1),
```
Comment (emschwartz) [inline api/src/models/payment.js]: Why does changing the order of these matter?

Comment (sharafian) [inline api/src/models/payment.js]: Sqlite supports the syntax where LIMIT is first, and postgres supports both.

**Comment by sharafian** [comment]
I'm sometimes getting an error here:

```
[api]    { Error: SQLITE_ERROR: near "from": syntax error
[api]        at Error (native)
[api]      errno: 1,
[api]      code: 'SQLITE_ERROR',
[api]      sql: 'update "Payments" set source_identifier = textcat(textcat(substring(source_account from \'https?://.*/ledger/accounts/(.*)\'), \'@\'), substring(source_account from \'https?://(.*)/ledger/accounts\')), destination_identifier = textcat(textcat(substring(destination_account from \'https?://.*/ledger/accounts/(.*)\'), \'@\'), substring(destination_account from \'https?://(.*)/ledger/accounts\'))' },
```

@vhpoet , do you know why the 12/16 migration script would cause this problem with sqlite? is there any way to make it compatible?

**Comment by vhpoet** [comment]
do we absolutely need sqlite support? it's adding an overhead, and doesn't let us use the awesomeness that postgresql provides.

**Comment by sharafian** [comment]
sqlite makes things _so_ much easier in a lot of cases though. For one, not requiring an entire postgres setup lets us bring the quick start guide down from 70 lines to 6. For another, it lets us write more effective unit tests for the ILP Kit without having to create and wipe a postgres database for every single test.

**Comment by adrianhopebailie** [comment]
I like sqlite with a file based DB for testing. I can setup accounts etc and then backup the file and go back to a known state each time I want to start testing.

**Comment by vhpoet** [comment]
ok well I can't merge this right now, will have to work on the transaction grouping (date_trunc) first.

**Comment by dappelt** [comment]
Any update on this? Having the ilp kit run on sqlite would be great for the integration tests.

**Comment by vhpoet** [comment]
@dappelt will work on it today

**Comment by vhpoet** [comment]
implemented in https://github.com/interledgerjs/ilp-kit/commit/a4e0a18a37c3132d60972cfe9bbaf5474db3d465. Note: since this is a quick fix so that @dappelt can run the integration tests, the history wouldn't work now.

**Pull request closed without merging**